src/components/browse/dataset/datasetPagination.tsx

```diff
@@ -1,30 +1,53 @@
-import React from 'react';
-import { Pagination } from 'react-bootstrap';
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import ReactPaginate from 'react-paginate'
+import { searchResponseModel } from "../../../models/dataset";
+import { getDatasetsSearchResp } from "../../../api/browse";
+import { facetFilterModel } from '../../../models/facets'
 
 interface dataSetPaginationProps {
-  dsCount : number
-  active : number
-  setActive: React.Dispatch<React.SetStateAction<number>>
+  dsCount: number;
+  setSearchResp: Dispatch<SetStateAction<searchResponseModel | null>>;
+  searchKeyword: string;
+  limit: number;
+  setLimit: any;
+  filterDict: facetFilterModel[];
 };
 
 const DatasetPagination = (props: dataSetPaginationProps) => {
+  const [pageCount, setpageCount] = useState(0);
 
-  const PaginationGenerator = (active : number = 1) => {
-    var items = []
-    for (let i = 1; i <= props.dsCount/10; i++) {
-      items.push(<Pagination.Item key={i} active={i === active} onClick={()=> props.setActive(i)}>{i}</Pagination.Item>)
-    }
+  useEffect(() => {
+    setpageCount(Math.ceil(props.dsCount / props.limit))
+  }, [props.dsCount, props.limit]);
 
-  return items
-}
+  const handlePageClick = async (data: any) => {
+    let skip = (data.selected) * props.limit;
+    getDatasetsSearchResp(props.setSearchResp, props.filterDict, props.searchKeyword, skip, props.limit)
+  };
 
   return (
-    <div>
-        <Pagination>
-            {PaginationGenerator(props.active)}
-        </Pagination>
+    <div className="container">
+      <ReactPaginate
```

Comment:
One thing we should fix is that when you change pages, the expanded datasets remain the same, rather than reset to all collapsed, or saving which were expanded and which not for each page..

Reply:
Lets go for a different tickets for these features/bugs